docs/source/data/data_catalog.md

```diff
@@ -667,6 +667,136 @@ You can use dataset factories to define a catch-all pattern which will overwrite
+
+"{default_dataset}":
+  type: pickle.PickleDataSet
+  filepath: data/01_raw/{default_dataset}.pickle
+```
+
+Running `kedro catalog rank` will result in the following output:
+
+```
+- preprocessed_{dataset_name}
+- processed_{dataset_name}
+- '{layer}.{dataset_name}'
+- '{default_dataset}'
+```
+
+#### Using `kedro catalog resolve`
+
+This command resolves dataset patterns in the catalog against any explicit dataset entries in the project pipeline. The resulting output contains all explicit dataset entries in the catalog and any dataset in the default pipeline that resolves some dataset pattern.
+
+To illustrate this, consider the following catalog file:
+
+```yaml
+companies:
+  type: pandas.CSVDataSet
+  filepath: data/01_raw/companies.csv
+
+reviews:
+  type: pandas.CSVDataSet
+  filepath: data/01_raw/reviews.csv
+
+shuttles:
+  type: pandas.ExcelDataSet
+  filepath: data/01_raw/shuttles.xlsx
+  load_args:
+    engine: openpyxl # Use modern Excel engine, it is the default since Kedro 0.18.0
+
+preprocessed_{name}:
+  type: pandas.ParquetDataSet
+  filepath: data/02_intermediate/preprocessed_{name}.pq
+
+"{default}":
+  type: pandas.ParquetDataSet
+  filepath: data/03_primary/{default}.pq
+```
+
+and the following pipeline in `pipeline.py`:
+
+```python
+def create_pipeline(**kwargs) -> Pipeline:
+    return pipeline(
+        [
+            node(
+                func=preprocess_companies,
+                inputs="companies",
+                outputs="preprocessed_companies",
+                name="preprocess_companies_node",
+            ),
+            node(
+                func=preprocess_shuttles,
+                inputs="shuttles",
+                outputs="preprocessed_shuttles",
+                name="preprocess_shuttles_node",
+            ),
+            node(
+                func=create_model_input_table,
+                inputs=["preprocessed_shuttles", "preprocessed_companies", "reviews"],
+                outputs="model_input_table",
+                name="create_model_input_table_node",
+            ),
+        ]
+    )
+```
+
+The resolved catalog output by the command will be as follows:
+
+```yaml
+companies:
+  filepath: data/01_raw/companies.csv
+  type: pandas.CSVDataSet
+model_input_table:
+  filepath: data/03_primary/model_input_table.pq
+  type: pandas.ParquetDataSet
+preprocessed_companies:
+  filepath: data/02_intermediate/preprocessed_companies.pq
+  type: pandas.ParquetDataSet
+preprocessed_shuttles:
+  filepath: data/02_intermediate/preprocessed_shuttles.pq
+  type: pandas.ParquetDataSet
+reviews:
+  filepath: data/01_raw/reviews.csv
+  type: pandas.CSVDataSet
+shuttles:
+  filepath: data/01_raw/shuttles.xlsx
+  load_args:
+    engine: openpyxl
+  type: pandas.ExcelDataSet
+```
+
+By default this is output to the terminal. However, if you wish to output the resolved catalog to a specific file, you can use the redirection operator `>`:
+
+```bash
+kedro catalog resolve > output_file.yaml
```

Comment:
This can then be picked up by the config loader, but then the pre-existing datasets that came from explicit entries in the catalog would need to be commented out so there's no duplicates. Maybe we could suggest this in the docs?  Not sure if this would make this part of the docs unnecessarily complicated for the users though so feel free to ignore this.

Reply:
I worry this might be a bit out of place, the beginning of this page in the docs does already talk about having different catalog files in different environments, it might get a bit messy to start explaining that here as well? I'm leaning against it but also curious if anyone else has any other opinions on this